geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/BaseCommand.java

```diff
@@ -137,6 +138,18 @@
     return OK_BYTES;
   }
 
+  protected boolean checkAndSetLastResultSentIfValid(
```

Comment:
The lastResult and sendResult are synchronized, does this needs to be synchronized?